static/src/javascripts/projects/commercial/modules/dfp/on-slot-render.ts

```diff
@@ -0,0 +1,100 @@
+import { adSizes } from '@guardian/commercial-core';
+import type { AdSizeString } from '@guardian/commercial-core';
+import { once } from 'lodash-es';
+import config from '../../../../lib/config';
+import mediator from '../../../../lib/mediator';
+import reportError from '../../../../lib/report-error';
+import { markTime } from '../../../../lib/user-timing';
+import { dfpEnv } from './dfp-env';
+import { emptyAdvert } from './empty-advert';
+import { getAdvertById } from './get-advert-by-id';
+import { renderAdvert } from './render-advert';
+
+const recordFirstAdRendered = once(() => {
+	markTime('Commercial: First Ad Rendered');
+});
+
+const reportEmptyResponse = (adSlotId: string, event: SlotRenderEndedEvent) => {
+	// This empty slot could be caused by a targeting problem,
+	// let's report these and diagnose the problem in sentry.
+	// Keep the sample rate low, otherwise we'll get rate-limited (report-error will also sample down)
+	if (Math.random() < 0.0001) {
```

Comment:
For your consideration:

```suggestion
	if (Math.random() < 1 / 10_000) {
```

It has a zero-perf cost, according to https://jsbench.me/x1ku71w7za/1